### ISSUE TITLE
How to calculate calibrationTransform?

### ISSUE THREAD

**Issue by Vincy-Ying** [herzig/bewegungsfelder#3]
Hello, I have some questions about CalculateCalibrationTransform() in SensorBoneLink.cs? First, "source " has been converted into "source.Transposed", and then, when getting  "CalibrationTransform", "source" has been converted into  "source.Transposed" again, so, its value is the initial value. Why do you make two transposed transforms? Why do you use "target" to multiply it?

**Comment by herzig** [comment]
Hi Vincy-Ying, Thanks for pointing that out.
I agree, all the calculations in CalculateCalibrationTransform seem to be completely unnecessary. It's been a long time since I wrote that Code and I have to admit I didn't understand fully what I was doing. That whole calibration transform part seems really sketchy looking at it now and it may very well be that it doesn't work properly. When I used the system, I always tried to align IMU axes with body axes as nicely as possible, thus not requiring the calibration, as it is quite tedious to do the way it is implemented now even if the code would work properly. 

So please feel free to fix that section and let me know if you come up with a better solution. For me the project is on-hold for the time being, and I most likely won't be able to work on this in a foreseeable future.

**Comment by Vincy-Ying** [comment]
> Hi Vincy-Ying, Thanks for pointing that out.
> I agree, all the calculations in CalculateCalibrationTransform seem to be completely unnecessary. It's been a long time since I wrote that Code and I have to admit I didn't understand fully what I was doing. That whole calibration transform part seems really sketchy looking at it now and it may very well be that it doesn't work properly. When I used the system, I always tried to align IMU axes with body axes as nicely as possible, thus not requiring the calibration, as it is quite tedious to do the way it is implemented now even if the code would work properly.
> So please feel free to fix that section and let me know if you come up with a better solution. For me the project is on-hold for the time being, and I most likely won't be able to work on this in a foreseeable future.

Thank you for your reply. I have tried to run the Bewegungsfelder.exe and loaded bvh file into it, the application always failed and exited. I donnot know the reason, can you  help me with it? Do I need to load the bvh file recorded by your sensors, not any other bvh files?
Can you share any referenced materials or similar programs about how to construct the motion capture application? (Email: 2468258590@qq.com)
Looking forward to your reply,thanks!

**Comment by herzig** [comment]
Yes, it should work with third-party bvh files, it's tested with a couple files I found online. If you can provide the file, I can try and have a look.

**Comment by Vincy-Ying** [comment]
Thank you for your reply.&nbsp;
The bvh file is attached in the email. This bvh file is found online, when I run Bewegungsfelder.exe in your package bewegungsfelder_v1.1.0 （because I cannot compile your github program on my local pc），
, and then click "File"-"Load Bvh", and choose this bvh file, it always crashes.


Best wishes，
Vincy-ying




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "herzig/bewegungsfelder"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年9月13日(星期天) 晚上11:34
收件人:&nbsp;"herzig/bewegungsfelder"<bewegungsfelder@noreply.github.com&gt;;
抄送:&nbsp;"周一"<2468258590@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [herzig/bewegungsfelder] How to calculate calibrationTransform? (#3)





 
Yes, it should work with third-party bvh files, it's tested with a couple files I found online. If you can provide the file, I can try and have a look.

**Comment by Vincy-Ying** [comment]
> Yes, it should work with third-party bvh files, it's tested with a couple files I found online. If you can provide the file, I can try and have a look.

Thanks, I have sent email to you and the bvh file has been attached.

**Comment by herzig** [comment]
> Thanks, I have sent email to you and the bvh file has been attached.

Hi, sorry for not getting back to you earlier.I have not seen any BVH file in my emails.
If this is still an issue, can you attach it here or upload it somewhere else?

**Comment by Vincy-Ying** [comment]
Thank you for your reply. I has solved that issue. Is your research direction about wearable system or motion capture, there are some key points I want to discuss with you, thanks!





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "herzig/bewegungsfelder"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年10月9日(星期五) 下午5:01
收件人:&nbsp;"herzig/bewegungsfelder"<bewegungsfelder@noreply.github.com&gt;;
抄送:&nbsp;"是初"<2468258590@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [herzig/bewegungsfelder] How to calculate calibrationTransform? (#3)





  
Thanks, I have sent email to you and the bvh file has been attached.
  
Hi, sorry for not getting back to you earlier.I have not seen any BVH file in my emails.
 If this is still an issue, can you attach it here or upload it somewhere else?

**Comment by BestPolarBear** [comment]
@Vincy-Ying hello,我也在做动作捕捉，mpu6050+esp8266，在unity环境中，我已经给你发了邮件，遇到一些问题。方便时候交流一下。
